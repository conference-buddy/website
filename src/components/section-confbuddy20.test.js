import React from "react";
import { screen, render, cleanup } from "@testing-library/react";
import ConfBuddy20Section from "./section-confbuddy-20";

describe("<ConfBuddy20Section />", () => {
  let component;
  beforeAll(() => {
    component = render(<ConfBuddy20Section />);
  });

  afterAll(cleanup);

  it("shows a header", () => {
    const header = screen.getAllByRole("heading", {
      name: "Conference Buddy 2.0.",
    });

    expect(header[0]).toBeVisible();
    expect(header[1]).toBeVisible();
  });

  it("shows a form", () => {
    const form = screen.getByRole("form");

    expect(form).toBeVisible();
    expect(form).toHaveAttribute(
      "action",
      "https://buttondown.email/api/emails/embed-subscribe/conferencebuddy"
    );
  });

  it("shows an input for email address", () => {
    const input = screen.getByRole("textbox", { name: "Enter your email" });

    expect(input).toBeVisible();
  });

  it("shows submit button to subscribe", () => {
    const submitButton = screen.getByRole("button", { name: "Subscribe" });

    expect(submitButton).toBeEnabled();
  });

  it("shows all elements", () => {
    expect(component.asFragment()).toMatchSnapshot();
  });
});
