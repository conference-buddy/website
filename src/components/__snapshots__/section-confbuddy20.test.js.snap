// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfBuddy20Section /> shows all elements 1`] = `
<DocumentFragment>
  <div
    class="container-fluid position-relative bg-bordered-wave"
  >
    <div
      class="p-content-section"
      id="onPageLinkId-section-about"
    >
      <div
        class="container"
      >
        <div
          class="d-none d-sm-block text-center"
        >
          <h2
            class="display-4"
          >
            Conference Buddy 2.0.
          </h2>
        </div>
        <div
          class="d-block d-sm-none text-center "
        >
          <h2>
            Conference Buddy 2.0.
          </h2>
        </div>
        <div
          class="pb-5 row justify-content-center"
        >
          <div
            class="col col-lg-8"
          >
            <h3
              class="text-center"
            >
              ✨ Webapp in progress ✨
            </h3>
            <p>
              I'm actively working a release of a first MVP for a Conference Buddy web app! You can follow along the progress in the 
              <a
                href="https://github.com/conference-buddy/conference-buddy-web-app#readme"
              >
                github
              </a>
              !
            </p>
            <p>
              If you want to be informed about the most important milestones - alpha release and, of course, the real release - you can sign up to my mailing list.
            </p>
          </div>
        </div>
        <div
          class="row col justify-content-center"
        >
          <form
            action="https://buttondown.email/api/emails/embed-subscribe/conferencebuddy"
            method="post"
            name="email-subscribe"
            target="popupwindow"
          >
            <label
              class="sr-only"
              for="bd-email"
            >
              Enter your email
            </label>
            <div
              class="input-group"
            >
              <input
                class="form-control"
                id="bd-email"
                name="email"
                placeholder="Enter your email"
                type="email"
              />
              <div
                class="input-group-append"
              >
                <input
                  class="btn btn-outline-secondary"
                  type="submit"
                  value="Subscribe"
                />
              </div>
            </div>
            <p
              class="text-center mt-2"
            >
              <a
                href="https://buttondown.email"
                rel="noreferrer"
                target="_blank"
              >
                Powered by Buttondown.
              </a>
            </p>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
